### DISCUSSION TITLE
[Lg webOS - piccap] Rename LUT after calibration

### DISCUSSION THREAD

**Discussion by GMD87** [awawa-dev/HyperHDR#730] [Q&A]
Hello everyone,
I am new to HyperHDR but I have a working setup (still v19.0.0.0) on my LG TV.
I wanted to go to the next step and perform a LUT calibration via web interface.
I used Firefox for the calibration and everything was successful, but I noticed that the generated LUT is saved as "lut_lin_tables.3d", while my setup (webOs - Piccap) uses the LUT named "Flat_lut_lin_tables.3d".

Unfortunately, I have no idea how to change the file name on webOs. Can anyone give me some guidance on this?

Will it still be necessary to rename the generated LUT in HyperHDR v20.0.0.0? If so, I might be better off waiting?

Thanks in advance!

**Comment by awawa-dev** [comment]
Hi
v19 and v20 behave the same way. You can simply delete Flat_lut_lin_tables.3d (it's special first-choice, 'compressed' smaller LUT only for proto/flatbuffers) or rename it to eg old_Flat_lut_lin_tables.3d You don't need to rename lut_lin_tables.3d.

**Reply by GMD87** [reply]
Hi Awawa, thank you for your response.
Would you have any way to give me some pointers on how to navigate to the folder where the LUTs are stored and delete/rename the file you pointed me to? Unfortunately my knowledge of webOS, ssh and HyperHDR is very limited.

I see from the logs that HyperHDR uses a flatbuffers, probably because I use Piccap installed directly on my TV as a grabber.

If I remember correctly, after logging in via SSH, I have to use the command 
`cd /home/root/.hyperhdr/`
to get to the folder where the LUTs are stored, and then use the command 
`mv Flat_lut_lin_tables.3d OLDFlat_lut_lin_tables.3d`
to change the filename. 
All correct? 

Or is it better in my case to continue using the Flat_lut_lin_tables.3d?

**Reply by awawa-dev** [reply]
I dont know where default/installed LUT (Flat_lut_lin_tables.3d) is stored on  WebOS, on other systems it's not a user home directory.
But you should have a full path to Flat_lut_lin_tables.3d in logs for sure, if tone mapping is enabled. The command to rename file (mv) seems OK

**Reply by GMD87** [reply]
After relaunching the calibration tool, the results look a bit weird. 
https://pastebin.com/bMN9pTAF

It seems that the colors after calibration are worse than the original ones and the " mean error" values are particularly high.
I performed the calibration directly from a pc (Windows 10) connected to the LG TV, Filmmaker mode, color depth (from graphics card settings) at 8bit and RGB 4:4:4 format. Auto HDR at 35.

Looking at the first lines of the log it also seems to me that HyperHDR for webOS keeps looking for the LUT for flatbuffer.
I haven't yet tried renaming the file flat_lut_lin_tables.3d though.

**Reply by awawa-dev** [reply]
Yes, this doesn't make sense. The calibration received such colors on input and failed to improve them.

```
2024-01-15T21:28:34.380Z [CALIBRATOR] (LutCalibrator.cpp:789) --------------------- Captured colors starts ---------------------
2024-01-15T21:28:34.380Z [CALIBRATOR] (LutCalibrator.cpp:790) Red: (255,0,0) => (101, 108, 27)
2024-01-15T21:28:34.380Z [CALIBRATOR] (LutCalibrator.cpp:791) Green: (0,255,0) => (22, 77, 90)
2024-01-15T21:28:34.380Z [CALIBRATOR] (LutCalibrator.cpp:792) Blue: (0,0,255) => (99, 100, 83)
```

Did you set high resolution in Piccap for calibration and did you noticed in the live preview if some browser's scrollbars/menu appeared? v20 has additional protection to avoid it. The calibration "mosaic" should take all the screen. No scrollbars, no menu or anything else should be visible.

**Reply by GMD87** [reply]
I've set the resolution to 1920x1080. No scrollbars or anything other than the mosaic visible. I Still run v19

**Reply by awawa-dev** [reply]
maybe re-calibrate? on the second computer, you can open 'live preview' and take a screenshot of what the calibration mosaic looks like during calibration.

**Reply by awawa-dev** [reply]
BTW I forgot. Did you disable HDR tone mapping for flatbuffer? It's very important and I'm not sure if HyperHDR will check it on webos.

**Reply by GMD87** [reply]
Ok, I'm dumb, there was a scrollbar on the right.
This is the new log: 
https://pastebin.com/ebYqFuzA

I've made the calibration before reading your last message, so I've not disabled HDR tone mapping

**Reply by awawa-dev** [reply]
I've seen similar results on webos. Unfortunately, HyperHDR does not have full control over calibration, because, for example, the YUV to RGB conversion is managed by Piccap and HyperHDR cannot choose the optimal strategy. Another thing is to use undocumented reverse engineering functions to make the capturing work at all.

**Reply by GMD87** [reply]
Seems good to me, non perfect, but a starting point. Do you know if v20 cam produce better results for my case?

Last question: from the logs i see  that the LUT file is saved as /home/root/.hyperhdr/lut_lin_tables.3d
but the flatbuffer contunes to search for the flat_lut. 
Is it safe to change the name (from lut_lin... to Flat_lut_lin...) from the directory above? 
That should also resolve the error from the first line in the log posted
`[FLATBUFSERVER] LUT file is not found here: /home/root/.hyperhdr/flat_lut_lin_tables.3d`

Anyway, thanks for the support!

**Reply by awawa-dev** [reply]
This is just a warning because flat_lut_lin_tables.3d is the first choice name for Flatbuffer LUT. If not found, HyperHDR will search for lut_lin_tables.3d. In the following lines you should have confirmation that it loaded the correct LUT. I wouldn't change its name, because such file may interfere if you want to perform another calibration.

v20 has one fix but for SDR calibration and nothing will change here,
